source/frontend/src/features/mapSideBar/acquisition/tabs/expropriation/form5/ExpropriationForm5.tsx

```diff
@@ -0,0 +1,103 @@
+import { Formik, FormikHelpers, FormikProps } from 'formik';
+import React from 'react';
+import { Col, Row } from 'react-bootstrap';
+import styled from 'styled-components';
+
+import { Button } from '@/components/common/buttons';
+import { ContactInputContainer } from '@/components/common/form/ContactInput/ContactInputContainer';
+import ContactInputView from '@/components/common/form/ContactInput/ContactInputView';
+import FormItem from '@/components/common/form/FormItem';
+import { SectionField } from '@/components/common/Section/SectionField';
+import { RestrictContactType } from '@/components/contact/ContactManagerView/ContactFilterComponent/ContactFilterComponent';
+import FilePropertiesTable from '@/components/filePropertiesTable/FilePropertiesTable';
+import { Api_AcquisitionFile } from '@/models/api/AcquisitionFile';
+import { Api_PropertyFile } from '@/models/api/PropertyFile';
+
+import { ExpropriationForm5Model } from '../models';
+import { ExpropriationForm5YupSchema } from './ExpropriationForm5YupSchema';
+
+export interface IExpropriationForm5Props {
+  acquisitionFile: Api_AcquisitionFile;
+  onGenerate: (values: ExpropriationForm5Model) => Promise<void>;
+}
+
+export const ExpropriationForm5: React.FC<IExpropriationForm5Props> = ({
+  acquisitionFile,
+  onGenerate,
+}) => {
+  const onSubmit = async (
+    values: ExpropriationForm5Model,
+    formikHelpers: FormikHelpers<ExpropriationForm5Model>,
+  ) => {
+    try {
+      await onGenerate(values);
+    } finally {
+      formikHelpers?.setSubmitting(false);
+    }
+  };
+
+  const onCancelClick = (formikProps: FormikProps<ExpropriationForm5Model>) => {
+    formikProps.resetForm();
+  };
+
+  const onGenerateClick = (formikProps: FormikProps<ExpropriationForm5Model>) => {
+    formikProps.setSubmitting(true);
+    formikProps.submitForm();
+  };
+
+  return (
+    <Formik<ExpropriationForm5Model>
+      enableReinitialize
```

Comment:
is this necessary?

Reply:
Not sure - I have used it on every formik form I create. Is there any harm in setting it to **true**?